src/standard/x-styling.html

```diff
@@ -144,9 +144,18 @@
             _scopeSelector: this._scopeSelector,
             _styleProperties: this._styleProperties
           };
-          scopeData.key.customStyle = {};
-          this.mixin(scopeData.key.customStyle, this.customStyle);
-          scope._styleCache.store(this.is, info, scopeData.key, this._styles);
+          // the scope cache does not evaluate if @media rules, :host(), or :host-context() rules defined in this element have changed
+          // therefore, if we detect those rules, we opt-out of the scope cache
+          var scopeCacheable = this._styles.every(function(s) {
+            return s._cacheable;
+          });
```

Comment:
Since we don't actually care about whether individual styles are cacheable, only whether _all_ styles for this element are cacheable, it would be good to move the `_notCachable` flag up to the element level so we can get rid of this instance-time loop. Seems like the `_notCacheable` flag could just hang off of `_styles`, perhaps, since it looks like [this method](https://github.com/Polymer/polymer/pull/3702/files#diff-5c83131b6cb1b0c603817554db3c5399R30) also gets `styles`?